tracer/src/Datadog.Trace/ClrProfiler/NativeCallTargetUnmanagedMemoryHelper.cs

```diff
@@ -0,0 +1,403 @@
+﻿// <copyright file="NativeCallTargetUnmanagedMemoryHelper.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Datadog.Trace.ClrProfiler;
+
+internal static class NativeCallTargetUnmanagedMemoryHelper
+{
+    private const int SizeOfMemorySegment = 750 * 1024; // 750Kb
+    private static readonly int SizeOfPointer = Marshal.SizeOf(typeof(IntPtr));
+    private static readonly List<IntPtr> Segments;
+    private static IntPtr _currentSegment;
+    private static int _segmentOffset;
+
+    static NativeCallTargetUnmanagedMemoryHelper()
+    {
+        Segments = new(5);
+        CreateSegment();
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static void CreateSegment()
+    {
+        _currentSegment = Marshal.AllocCoTaskMem(SizeOfMemorySegment);
```

Comment:
Any reason why you're using AllocCoTaskMem rather than AllocHGlobal?

Reply:
Just to copy the behavior of the BCL marshallers, for example:

- https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/Utf8StringMarshaller.cs

- https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshalling/ArrayMarshaller.cs